releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java

```diff
@@ -79,7 +80,50 @@ public static List<String> runPostGenerationAndPublication(
             Multimap<String, ReleaseNotesMessage> releaseNotes, CliOptions cliOptions)
             throws IOException, TemplateException {
         runPostGeneration(releaseNotes, cliOptions);
-        return runPostPublication(cliOptions);
+        final List<String> errors = new ArrayList<>();
+        errors.addAll(validateNotes(releaseNotes, cliOptions));
+        errors.addAll(runPostPublication(cliOptions));
+        return errors;
+    }
+
+    /**
+     * Validate that pom version matches issues labels in release notes.
+     *
+     * @param releaseNotes map of release notes messages.
+     * @param cliOptions command line options.
+     * @return list of notes validation errors.
+     */
+    private static List<String> validateNotes(Multimap<String,
+                                                    ReleaseNotesMessage> releaseNotes,
+                                                    CliOptions cliOptions) {
+        final String releaseVersion = cliOptions.getReleaseNumber();
+        final long amountOfNumbers = getAmountOfNumbersInReleaseVersion(releaseVersion);
+
+        final boolean containsNewOrBreakingCompatabilityLabel
+            = releaseNotes.containsKey(Constants.NEW_FEATURE_LABEL)
+            || releaseNotes.containsKey(Constants.NEW_MODULE_LABEL)
+            || releaseNotes.containsKey(Constants.BREAKING_COMPATIBILITY_LABEL);
+
+        final List<String> errors = new ArrayList<>();
+
+        if (amountOfNumbers == 2 && containsNewOrBreakingCompatabilityLabel
+            || amountOfNumbers == 3 && !containsNewOrBreakingCompatabilityLabel) {
+            errors.add("Validation of release notes failed. "
+                + "There is a mismatch between release version and issues labels.");
```

Comment:
Was wondering whether we should throw this generic message or explain a bit in depth what we are expecting depending on `releaseVersion`. Something like
```java
if (amountOfNumbers == 2 && containsNewOrBreakingCompatabilityLabel) {
    errors.add(String.format("Validation of release notes failed. Release version contains two numbers(%s) and new or breaking compatibility labels.", releaseVersion))
} else if (amountOfNumbers == 3 && !containsNewOrBreakingCompatabilityLabel) {
    errors.add(String.format("Validation of release notes failed. Release version contains three numbers(%s) and no new or breaking compatibility labels.", releaseVersion))
}
```

Reply:
It is very bad practice to do logic based on exceptions.
We can create class to keep violation message, or simply keep list of strings - errors

Reply:
Changed to keep list of strings - errors. Done.